lib/runner/connection.rb

```diff
@@ -208,4 +220,51 @@ def handle_timeout(_event)
     # The runner management stopped the execution as the permitted execution time was exceeded.
     # We set the status here and wait for the connection to be closed (by the runner management).
   end
+
+  # The methods below are inspired by the Sentry::Net:HTTP class
+  # and adapted to the Websocket protocol running with EventMachine.
+
+  def clone_sentry_hub_from_sentry_span(sentry_span)
+    Thread.current.thread_variable_set(Sentry::THREAD_LOCAL, sentry_span.transaction.hub) if sentry_span
+  end
+
+  def sentry_trace_header(sentry_span)
+    return {} unless sentry_span
+
+    http_headers = {}
+    client = Sentry.get_current_client
+
+    trace = client.generate_sentry_trace(sentry_span)
+    http_headers[Sentry::SENTRY_TRACE_HEADER_NAME] = trace if trace
+
+    baggage = client.generate_baggage(sentry_span)
+    http_headers[Sentry::BAGGAGE_HEADER_NAME] = baggage if baggage.present?
+
+    {
+      headers: http_headers,
+    }
+  end
+
+  def end_sentry_span(sentry_span, event)
+    return unless sentry_span
+
+    sentry_span.set_description("WebSocket #{@socket.url}")
```

Comment:
As the `@socket.url` contains the runner id and execution id the Sentry Table `Suspect Spans` cannot correctly aggregate the spans (but counts every span individually). We may avoid those dynamic values

Reply:
Thanks for identifying that. I see three possibilities to continue with:

1. Remove the Runner ID (path) and Execution ID (Query)
2. Only remove the Execution ID (Query); that's how Sentry works for the patch route
3. Remove nothing, as the query parameter would only be removed to ensure that no personal identifiable information is sent (which is not the case here).

I compared the WebSocket spans with the `PATCH files` routes, and for the latter (where we don't adjust anything so far), the "Suspect Spans" are also not grouped correctly (only for the runners), not overall for the API route. Did you notice that as well?

Reply:
Yes, you are right, this problem does also affect the PATCH route.
Although, I see the problem in Sentry and not in our usage I would think about fixing it to have the useful overview of which span uses the most time.
Here, we can fix it by your option 1.
For the other routes by code such as:
``` go
func initSentry(options *sentry.ClientOptions) {
	options.BeforeSendTransaction = func(event *sentry.Event, _ *sentry.EventHint) *sentry.Event {
		fields := strings.Split(strings.Split(strings.Fields(event.Transaction)[1], "?")[0], "/")
		event.Transaction = fields[len(fields)-1]
		return event
	}
	if err := sentry.Init(*options); err != nil {
		log.Errorf("sentry.Init: %s", err)
	}
}
```
What do you think?